zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -39,14 +40,39 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
     version = BuildConfig.KOTLIN_PLUGIN_VERSION,
   )
 
+  override fun apply(target: Project) {
+    super.apply(target)
+
+    // If the build has compileDevelopmentExecutableKotlinJs, create developmentExecutableZipline.
+    // Symmetrically for production.
+    target.tasks.whenTaskAdded { task ->
+      if (task !is KotlinJsCompile) {
+        return@whenTaskAdded
+      }
+      val compileType = when (task.name) {
+        "compileDevelopmentExecutableKotlinJs" -> "developmentExecutable"
+        "compileProductionExecutableKotlinJs" -> "productionExecutable"
+        else -> return@whenTaskAdded
+      }
+      target.tasks.create("${compileType}Zipline", ZiplineCompileTask::class.java) { createdTask ->
```

Comment:
`create` forces eager configuration of the task. Use `register` so that it's lazy. Syntax should mostly be identical otherwise.

Reply:
Fixed.